src/domain/FlowRunner.ts

```diff
@@ -545,8 +551,13 @@ export class FlowRunner implements IFlowRunner, IFlowNavigator, IPromptBuilder {
     )
 
     interactions.push(richCursor.interaction)
+
+    // step cursor forward with context
     ctx.cursor = this.dehydrateCursor(richCursor)
 
+    // inflate prompt references
```

Comment:
Note, if this comment is needed, it means the function name is not clear enough.